api/collections/serializers.py

```diff
@@ -94,3 +94,76 @@ def get_absolute_url(self, obj):
                 'node_link_id': obj._id
             }
         )
+
+
+class SingleLinkedNode(JSONAPIRelationshipsSerializer):
+    id = ser.CharField(source='node._id', required=False, allow_null=True)
+    type = TypeField(required=False, allow_null=True)
+
+    class Meta:
+        type_ = 'linked_nodes'
+
+
+class CollectionLinkedNodesRelationshipSerializer(ser.Serializer):
+
+    data = ser.ListField(child=SingleLinkedNode())
+    links = LinksField({'self': 'get_self_url',
+                        'html': 'get_related_url'})
+
+    def get_self_url(self, obj):
+        return obj['self'].linked_nodes_self_url
+
+    def get_related_url(self, obj):
+        return obj['self'].linked_nodes_related_url
+
+    class Meta:
+        type_ = 'linked_nodes'
+
+    def update(self, instance, validated_data):
+        collection = instance['self']
+        auth = get_user_auth(self.context['request'])
+
+        new_pointer_ids = [val['node']['_id'] for val in validated_data['data']]
+        pointers_to_delete = [pointer for pointer in instance['data'] if pointer.node._id not in new_pointer_ids]
+        pointers_to_add = [node_id for node_id in new_pointer_ids if node_id not in [pointer.node._id for pointer in instance['data']]]
```

Comment:
Nested list comprehensions are not my thing. Also there's a lot of similarity between this and create. Can you maybe break these out into functions to DRY it up? Think about if this is essentially going to be the same logic between users of these relationship fields. If you can feed in a couple of lists, then people won't have to try to replicate the logic when they apply relationship self links to other fields.